platforms/core-runtime/base-services/src/main/java/org/gradle/internal/service/DefaultServiceRegistry.java

```diff
@@ -676,21 +707,30 @@ public final synchronized void stop() {
     private static abstract class SingletonService extends ManagedObjectServiceProvider {
         private enum BindState {UNBOUND, BINDING, BOUND}
 
-        final Type serviceType;
-        final Class<?> serviceClass;
+        protected final List<? extends Type> rawDeclaredServiceTypes;
+        private final List<Class<?>> unwrappedDeclaredServiceTypes;
```

Comment:
💭  Changing these types is the key part of the refactoring.

Reply:
Agreed, `raw` can be confusing in this context. A lot of code around already mixes "type" and "class" terms interchangeably, so the intention here was to make it more explicit.

Will polish in a follow-up